Documentation/AutoRetry.md

```diff
@@ -0,0 +1,38 @@
+# Auto retry
+----
+
+## Auto refresh token
+
+Using this property, you can tell your code how to renew the token. Typically, this closure will contain code to perform the "RefreshToken" request, parse the response, update the token stored in your NetworkStack, and emit a Observable.just() to tell when it's done.
+
+```swift 
+public var renewTokenHandler = () -> Observable<Void>
+```
+
+With this property you can renew your token automatically. This closure will be called when: 
+
+- You're authenticated
+- But you received an 401 error for a request
+- That renewTokenHandler closure is not nil
+
+## Renew Authent if needeed
+
+Compose with this optional property when you init NetworkStack:
+
+```swift 
+public var askCredential: AskCredential?
+```
+
+If you set this property you can relaunch your authenticate workflow automatically. This property is a struct that encapsulate a closure to handle re-authe,tication process and a condition to know when trigger the closure.
```

Comment:
`s/authe,tication/authentication/`

Isn't this property `askCredential` supposed just to let you present a Login screen for example… but then DON'T retry the request after that (which is the main difference between this and `renewToken`? So not sure if "you can relaunch your authenticate workflow automatically" is correct